Lib/tkinter/simpledialog.py

```diff
@@ -260,7 +260,15 @@ def __init__(self, title, prompt,
                  parent = None):
 
         if not parent:
-            parent = tkinter._default_root
+            if tkinter._default_root:
+                parent = tkinter._default_root
+            elif tkinter._support_default_root:
+                parent = tkinter.Tk()
+                parent.withdraw()
```

Comment:
If a new Tk instance is being created, I am of the opinion it should be explicitly deleted once we are done with the dialog (we shouldn't rely on the GC to clean it).

Reply:
On second thought, would it be better to mimic tkinter.commondialog (for consistency), which doesn't withdraw or delete a new window?

Reply:
I thought about that too. It was a few months ago, but I remember to have compared with what is done elsewhere and came to the same conclusion.